spec/lcm/integration/user_provisioning_end_to_end_spec.rb

```diff
@@ -0,0 +1,172 @@
+          project_id: c.pid,
+          position: 'developer'
+        }
+      end
+    end.flatten
+    Support::S3Helper.upload_file(ConfigurationHelper.csv_from_hashes(@user_data), Support::S3Helper::USERS_KEY)
+
+    @test_context = {
+      project_id: @projects.first.pid, # this doesn't really matter for the runtime
+      config: LcmConnectionHelper.environment,
+      s3_bucket: Support::S3Helper::BUCKET_NAME,
+      s3_endpoint: Support::S3Helper::S3_ENDPOINT,
+      s3_key: Support::S3Helper::USER_FILTERS_KEY,
+      sync_mode: 'sync_multiple_projects_based_on_pid',
+      data_product: 'default',
+      users_brick_input: {
+        s3_bucket: Support::S3Helper::BUCKET_NAME,
+        s3_endpoint: Support::S3Helper::S3_ENDPOINT,
+        s3_key: Support::S3Helper::USERS_KEY
+      }
+    }
+  end
+
+  describe '1 - initial MUF setup' do
+    before(:all) do
+      @user_data.map do |u|
+        @domain.add_user(login: u[:login]) unless @domain.users l
+      end
+
+      label, value = uniq_label_with_value @projects.first
+      @mufs = @user_data.map do |u|
+        m = {}
+        m[:login] = u[:login]
+        m[:project_id] = u[:project_id]
+        m[label.title] = value
+        m
+      end.flatten
+
+      Support::S3Helper.upload_file(ConfigurationHelper.csv_from_hashes(@mufs), Support::S3Helper::USER_FILTERS_KEY)
+    end
+
+    it 'setups the MUFs' do
+      BrickRunner.user_filters_brick context: @test_context, template_path: @ufb_template_path
+      @projects.each do |p|
+        filters = p.user_filters.to_a
+        expect(filters.length).to eq $user_amount
+        @user_data.each do |u|
+          expect(filters.find { |f| f.related.email == u[:login] })
+        end
+      end
+    end
+  end
+
+  describe '2 - initial user setup' do
+    it 'setups the users' do
+      BrickRunner.users_brick context: @test_context, template_path: @ub_template_path
+      @projects.each do |p|
+        users = p.users.to_a
+        expect(users.length).to eq($user_amount + 1) # rubydev+admin is in the projects as well (as he is the admin)
+      end
+    end
+  end
+
+  describe 'modify the user input' do
+    it 'does it' do
+      user_data = @user_data.map do |c|
+        cdup = c.dup
+        cdup['position'] = 'manager'
+        cdup
+      end.flatten
+      Support::S3Helper.upload_file(ConfigurationHelper.csv_from_hashes(user_data), Support::S3Helper::USERS_KEY)
+    end
+  end
+
+  describe 'modify the MUF input' do
+    it 'does it' do
+      label, value = uniq_label_with_value @project.first
+
+      mufs = @mufs.map do |m|
+        mdup = m.dup
+        mdup[label.title] = value
+        mdup
+      end
+
+      Support::S3Helper.upload_file(ConfigurationHelper.csv_from_hashes(mufs), Support::S3Helper::USER_FILTERS_KEY)
+    end
+  end
+
+  describe '3 - subsequent MUF setup' do
+    it 'checks if they appeared after adding users to project' do
+      @projects.each do |p|
```

Comment:
Identical code found in user_provisioning_end_to_end_spec.rb:101, user_provisioning_end_to_end_spec.rb:148 (mass = 124)